### PR TITLE
feat: Indexer support query with order

### PR DESCRIPTION
**Pull request by bitrocks**

### CODE REVIEW

File: packages/indexer/lib/index.js [hunk 1]
```diff
@@ -194,6 +195,9 @@ class CellCollector {
     if (type && typeof type === "object") {
       validators.ValidateScript(type);
     }
+    if (order !== "asc" && order != "desc") {
```
Comment (classicalliu) [inline packages/indexer/lib/index.js]:
```suggestion
    if (order !== "asc" && order !== "desc") {
```

File: packages/indexer/lib/index.js [hunk 2]
```diff
@@ -329,12 +339,17 @@ class TransactionCollector {
       this.type = type;
     }
 
+    if (order !== "asc" && order != "desc") {
```
Comment (classicalliu) [inline packages/indexer/lib/index.js]:
```suggestion
    if (order !== "asc" && order !== "desc") {
```

File: packages/sql-indexer/lib/index.js [hunk 3]
```diff
@@ -580,6 +581,9 @@ class CellCollector {
     if (type && typeof type === "object") {
       validators.ValidateScript(type);
     }
+    if (order !== "asc" && order != "desc") {
```
Comment (classicalliu) [inline packages/sql-indexer/lib/index.js]:
```suggestion
    if (order !== "asc" && order !== "desc") {
```

File: packages/indexer/lib/index.js [hunk 4]
```diff
@@ -249,7 +254,11 @@ class CellCollector {
     } else {
       outPoints = typeOutPoints;
     }
-    return outPoints;
+    if (this.order === "desc") {
+      return outPoints.reverse();
```
Comment (xxuejie) [inline packages/indexer/lib/index.js]: Are we sure this is enough? Don't we need to do a query from the other direction at rocksdb side? I recall there are a lot of code to support order query in ckb-indexer

Comment (bitrocks) [reply packages/indexer/lib/index.js]: I first came out with the same way you suggesting here, which would query the rocksdb with order parameter via rust bindings directly, but it seems quite ok to reserve the default query results, which is already well-ordered by block number  in ascending order.

Comment (bitrocks) [reply packages/indexer/lib/index.js]: Yep, it seems `not` ok after further discussion. gonna fix it.

**Review by xxuejie — APPROVED** [review]

**Pull request merged**